### PR TITLE
Add support for reserved fields

### PR DESCRIPTION
**Pull request by hslatman**

### CODE REVIEW

File: cabfile/cabfile.go [hunk 1]
```diff
@@ -132,9 +186,9 @@ func New(r io.ReadSeeker) (*Cabinet, error) {
 	if (hdr.Flags&hdrPrevCabinet) != 0 || (hdr.Flags&hdrNextCabinet) != 0 {
 		return nil, errors.New("multi-part Cabinet files are unsupported")
 	}
-	if (hdr.Flags & hdrReservePresent) != 0 {
-		return nil, errors.New("Cabinet files with reserved fields are unsupported")
-	}
+
+	// skip abReserve by skipping ahead cbCFHeader bytes
+	ioutil.ReadAll(io.LimitReader(r, int64(hdr.CBCFHeader)))
```
Comment (pkern) [inline cabfile/cabfile.go:191]: I'd prefer if we wouldn't swallow the error here. And maybe there could be a meaningful error if we hit EOF as part of the seeking?

File: cabfile/cabfile.go [hunk 2]
```diff
@@ -117,9 +121,59 @@ func New(r io.ReadSeeker) (*Cabinet, error) {
 
 	// CFHEADER
 	var hdr cfHeader
-	if err := binary.Read(r, binary.LittleEndian, &hdr); err != nil {
-		return nil, fmt.Errorf("could not deserialize header: %v", err)
+	if err := binary.Read(r, binary.LittleEndian, &hdr.Signature); err != nil {
```
Comment (pkern) [inline cabfile/cabfile.go:124]: I briefly wondered if there's a more convenient way (alike encoding/json), but it looks like the answer is no, unrolling is just the straightforward way to handle this.

File: cabfile/cabfile.go (hunk 1 above)
Comment (hslatman) [reply cabfile/cabfile.go]: I've changed this to `io.ReadFull` in 99593de99f00a0280258504ac5b938e7f4810dd5 and the error check is included now.

**Comment by hslatman** [comment]
Is there a specific minimum Go version you're targeting? If it's 1.11 (and later), does it make sense to add `go.mod` to ease local testing?

**Comment by pkern** [comment]
I'd be open to another PR adding them.

**Review by pkern — APPROVED** [review]
LGTM

**Comment by hslatman** [comment]
Thank you!

File: cabfile/cabfile.go (hunk 1 above)
Comment (pschou) [reply cabfile/cabfile.go]: I like the update you suggest

Comment (pschou) [reply cabfile/cabfile.go]: Thank you!

**Pull request merged**